### ISSUE TITLE
Juniper admin , firewall & idps logs are not parsed correctly

### ISSUE THREAD

**Issue by imsidr** [splunk/splunk-connect-for-syslog#2643]
**Note: If your issue is not a bug or a feature request, please raise a support ticket through our support portal (Splunk.com > Support > Support Portal). This will help us resolve your issue more efficiently and provide you with better assistance. For more information on how to work with the Splunk Support, please refer to [this guide](https://splunk.my.site.com/customer/s/article/Working-with-Support-and-the-Support-Portal).**

**Was the issue replicated by support?** no
**What is the sc4s version ?** 3.27.0

**Which operating system (including its version) are you using for hosting SC4S?** Ubuntu 

**Which runtime (Docker, Podman, Docker Swarm, BYOE, MicroK8s) are you using for SC4S?** docker

**Is there a pcap available? If so, would you prefer to attach it to this issue or send it to Splunk support?** yes

**Is the issue related to the environment of the customer or Software related issue?** Software related issue

**Is it related to Data loss, please explain ?** No
Protocol? Hardware specs?
 
**Last chance index/Fallback index?** sc4s

**Is the issue related to local customization?** yes

**Do we have all the default indexes created?** yes

**Describe the bug**
Juniper admin , firewall & idps logs are not parsed correctly

**To Reproduce**
1. Go to '...' index=firewall or index=juniper*

**Comment by imsidr** [comment]
pcap attached to splunk support case # CASE [3621290]

**Comment by imsidr** [comment]
Hi @rjha-splunk @sbylica-splunk any update on this ?

**Comment by sbylica-splunk** [comment]
Hi @imsidr, looking into it, I will post an update once we have something more.

**Comment by sbylica-splunk** [comment]
@imsidr so we decided to add parsing logs with an RT_SYSTEM tag, with a sourcetype `juniper:legacy`. Does that seem good to you?

**Comment by imsidr** [comment]
Hi Szymon,
Are you saying all the logs would have juniper:legacy sourcetype?

//BR,Sid

**Comment by sbylica-splunk** [comment]
@imsidr yeah, logs with RT_SYSTEM tag would be assigned to a `juniper:legacy` sourcetype.

**Comment by imsidr** [comment]
But I can see none of them are parsed.. for example I see RT_FLOW logs are not parsed.
//BR,Sid

**Comment by sbylica-splunk** [comment]
@imsidr can you attach an example of RT_FLOW log that is not being parsed properly? I don't see any in the pcap file that was attached. 
Logs with RT_FLOW tag should be parsed correctly since we have a rule for that.

**Comment by imsidr** [comment]
I asked Ravi from support to setup a call to show this issue, can we connect on this ?
I am sure all sorts of logs were captured in both the pcaps , I am not sure how I capture it for a particular log.


//BR,Sid

**Comment by sbylica-splunk** [comment]
@imsidr Yeah, we can setup a call. What would be a possible time for it?

**Comment by imsidr** [comment]
Can we have it today ?
I am available for next 4hrs.

**Comment by sbylica-splunk** [comment]
@imsidr today probably not, could we arrange for it tomorrow or next week?

**Comment by imsidr** [comment]
Sure , we can connect tomorrow , I am available between 3-7 PM IST & 9-11PM IST

//BR,Sid

**Comment by sbylica-splunk** [comment]
Hey @imsidr, so we have two follow-up questions for now:
- Can we get a .pcap file with an example of logs with RT_FLOW tag, like the one we saw on a meeting with the customer?
- Were the logs getting parsed earlier? Was there any noticeable trigger point that led to this issue?

**Comment by imsidr** [comment]
Hi Szymon ,
I have attached the pcap in spluk support case.


//BR,Sid

**Comment by sbylica-splunk** [comment]
@imsidr thanks, could you also ask for a screenshot from the Splunk side? We would like to see the message and the sc4s_tags field.

**Comment by imsidr** [comment]
I have attached screenshots in the case.

**Comment by sbylica-splunk** [comment]
Hi @imsidr, maybe I missed it but was there any response to this question?
```Were the logs getting parsed earlier? Was there any noticeable trigger point that led to this issue?```

Also, a new version of SC4S with the fix for RT_SYSTEM logs was released, can the customer upgrade to this version?
https://github.com/splunk/splunk-connect-for-syslog/releases/tag/v3.33.0

**Comment by imsidr** [comment]
Hi Szymon,
It was being parsed when we were onboarding it using legacy method. Also we have Juniper Add-on installed with some additional configs from out side but it is not related to parsing.
Also the issue is not related to RT_SYSTEM only , none of the events are parsed.
But I will still go ahead and upgrade it.
//BR,Sid

**Comment by sbylica-splunk** [comment]
Hi @imsidr, So, it was being parsed correctly using the legacy method? Do we have a configuration/description of this method? Ok, let's see if updating fixes this issue, keep me updated.

**Comment by imsidr** [comment]
We are using syslog-ng relay to get the data and forward it to the splunk cloud, we have juniper TA(customized) which is doing all the parsing & extraction stuff , I will attach the copy on support case.


//BR,Sid

**Comment by imsidr** [comment]
@sbylica-splunk @sbylica-splunk we are seeing jinja2 module errors after upgrading to v 3.33 , the container is in restarting status , we already have jinja2 module installed , not sure why are we getting this error. Attaching sc4s file in the case.

**Comment by cwadhwani-splunk** [comment]
Hi @imsidr 
We looked into the case again and observed that the logs that are being sent to SC4S are framed logs, and we noticed that you are sending it to tcp/514. We recommend to send the framed logs to tcp/601, here the framed logs are handled correctly. I tried to reproduce this issue by sending your logs to tcp/514, but as soon as I sent it to tcp/601, it was working fine, all the logs were sent indivially.
Could you please try this and let me know if it helped you.

**Comment by cwadhwani-splunk** [comment]
Closing this issue as the above suggestion worked for the customer. Feel free to reach out to us if you have any further queries.
